src/liftdrag_plugin/liftdrag_plugin.cpp

```diff
@@ -294,99 +298,92 @@ void LiftDragPlugin::OnUpdate()
   {
-    cl = (-this->cla * this->alphaStall +
-          this->claStall * (this->alpha + this->alphaStall))
-         * cosSweepAngle;
+    cl = -this->cla * this->alphaStall +
+         this->claStall * (this->alpha + this->alphaStall);
     // make sure cl is still less than 0
     cl = std::min(0.0, cl);
   }
   else
-    cl = this->cla * this->alpha * cosSweepAngle;
-
-  // modify cl per control joint value
-  double controlAngle;
-  if (this->controlJoint)
   {
-#if GAZEBO_MAJOR_VERSION >= 9
-    controlAngle = this->controlJoint->Position(0);
-#else
-    controlAngle = this->controlJoint->GetAngle(0).Radian();
-#endif
-    cl = cl + this->controlJointRadToCL * controlAngle;
-    /// \TODO: also change cd
+    cl = this->cla * this->alpha;
   }
 
+  // modify cl per control joint value
+  cl = cl + this->cl_delta * controlAngle;
+
   // compute lift force at cp
-  ignition::math::Vector3d lift = cl * q * this->area * liftI;
+  ignition::math::Vector3d lift = cl * q * this->area * liftDirection;
 
-  // compute cd at cp, check for stall, correct for sweep
+  // compute cd at cp, check for stall
   double cd;
-  if (this->alpha > this->alphaStall)
+  if (fabs(this->alpha) > this->alphaStall)
   {
-    cd = (this->cda * this->alphaStall +
-          this->cdaStall * (this->alpha - this->alphaStall))
-         * cosSweepAngle;
+    cd = this->cd_alpha0 + this->cda * this->alphaStall +
+         this->cdaStall * (this->alpha - this->alphaStall);
   }
-  else if (this->alpha < -this->alphaStall)
+  else
   {
-    cd = (-this->cda * this->alphaStall +
-          this->cdaStall * (this->alpha + this->alphaStall))
-         * cosSweepAngle;
+    cd = this->cd_alpha0 + this->cda * fabs(this->alpha);
   }
-  else
-    cd = (this->cda * this->alpha) * cosSweepAngle;
 
   // make sure drag is positive
   cd = fabs(cd);
 
+  // Take into account the effect of control surface deflection angle to Cd. Contribution must be positive.
+  cd += fabs(this->cd_delta * controlAngle);
+
   // drag at cp
   ignition::math::Vector3d drag = cd * q * this->area * dragDirection;
 
-  // compute cm at cp, check for stall, correct for sweep
+  // compute cm at cp, check for stall
   double cm;
   if (this->alpha > this->alphaStall)
   {
-    cm = (this->cma * this->alphaStall +
-          this->cmaStall * (this->alpha - this->alphaStall))
-         * cosSweepAngle;
-    // make sure cm is still great than 0
+    cm = this->cm_alpha0 + this->cma * this->alphaStall +
+         this->cmaStall * (this->alpha - this->alphaStall);
+    // make sure cm is still greater than 0
     cm = std::max(0.0, cm);
   }
   else if (this->alpha < -this->alphaStall)
   {
-    cm = (-this->cma * this->alphaStall +
-          this->cmaStall * (this->alpha + this->alphaStall))
-         * cosSweepAngle;
+    cm = this->cm_alpha0 - this->cma * this->alphaStall +
+         this->cmaStall * (this->alpha + this->alphaStall);
     // make sure cm is still less than 0
     cm = std::min(0.0, cm);
```

Comment:
Does this assumption still hold with the defined `cm_alpha0` ?

Reply:
Previously, cm behavior was symmetric around zero, while now we start at cm_alpha0. Given the simplified aerodynamics we use in this plugin, assuming now it is symmetric around cm_alpha0 makes sense. I'll add this to the pull request.

Note that for current simulation models, it doesn't matter, since those all set cm to 0 under all conditions.

Reply:
In fact, what I can find indicates that past stall, cm is typically more negative than cm_alpha0 (positive angle of attack), [e.g.  this](https://ocw.mit.edu/courses/aeronautics-and-astronautics/16-01-unified-engineering-i-ii-iii-iv-fall-2005-spring-2006/fluid-mechanics/f19_fall.pdf). I'll improve on this too.